Understanding Harvester Resource Metrics Calculation (#52)
harvester/harvesterhci.io: kb/2024-01-23/reserved_resources_calculation.md

---

> title: Understanding Harvester Reserved Resources Calculation
> description: Know hwo the resources reseravation is made and calculated
> slug: harvester_reserved_resources_calculation

Reviewer: ```suggestion
title: Calculation of Reserved Resources in Harvester
description: Understand how reserved resource metrics are calculated.
slug: calculation_of_reserved_resources_in_harvester
```

Refer [Kubernetes resources requests and limits](https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/#requests-and-limits) for more details.

#### Calculation

The `Resource Capacity` is the base data.

The `Used` is from `Resource Usage`.

The `Reserved` is from `Resource Reservation`.

And the Web UI also does some final rendings.

### Storage Usage

The Longhorn is the default CSI of Harvester, it manages the storage.

#### Reserved Storage in Longhorn

The reserved percentage specifies the percentage of disk space that will not be allocated to the default disk on each new Longhorn node.

This setting only affects the default disk of a new adding node or nodes when installing Longhorn.

Defautl value: 30

Refer [Longhorn reserved disks](https://longhorn.io/docs/1.5.3/references/settings/#storage-reserved-percentage-for-default-disk) for more details.

Depends on the default disk size, you may adjust this value via [the embedded Longhorn UI](https://docs.harvesterhci.io/v1.2/troubleshooting/harvester/#access-embedded-rancher-and-longhorn-dashboards).

::: note

You are suggested to confirm with Harvester engineering team before change the settings.

> :::

Author: @jillian-maroket I updated it to `You are suggested to read Longhorn documents carefully before change the settings.`
